src/RenderingSystem.cpp

```diff
@@ -456,15 +456,30 @@ void RenderingSystem::setup_cameras() {
 
         glm::vec3 car_pos(transform[3][0], transform[3][1] + 0.5, transform[3][2]);
 
-        auto camera_position = glm::translate(transform, glm::vec3(0, 3, -8));
+        auto camera_position = glm::translate(transform, glm::vec3(0, 5, -12));
```

Comment:
Camera position is too high and too far back. When going full speed, the camera is too far back.

My camera position at glm::vec3(0, 3, -8) works really well with the rest of your changes, imo.